### PR TITLE
Docker Client is now a singleton

### PR DESCRIPTION
**Pull request by remmelt**
Fixes: https://github.com/ContainerSolutions/minimesos/issues/361

### CODE REVIEW

File: minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java [hunk 1]
```diff
@@ -103,7 +88,7 @@ private MesosCluster(String clusterId) {
         this.clusterId = clusterId;
         this.clusterConfig = new ClusterConfig();
 
-        List<Container> dockerContainers = dockerClient.listContainersCmd().exec();
+        List<Container> dockerContainers = DockerClientFactory.get().listContainersCmd().exec();
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java]: * should we have only one `DockerClient dockerClient = DockerClientFactory.get()` call per method and then re-use the local variable?
* what is the difference between `DockerClientFactory.get()` and the previous `DockerClientFactory.build()`?

File: minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java [hunk 2]
```diff
@@ -139,13 +124,13 @@ private MesosCluster(String clusterId) {
                         }
                         break;
                     case "marathon":
-                        this.containers.add(new Marathon(dockerClient, this, uuid, containerId));
+                        this.containers.add(new Marathon(this, uuid, containerId));
                         break;
                     case "consul":
-                        this.containers.add(new Consul(dockerClient, this, uuid, containerId));
+                        this.containers.add(new Consul(DockerClientFactory.get(), this, uuid, containerId));
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java]: * Does `Consul` take docker client as a parameter?

File: minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java [hunk 3]
```diff
@@ -139,13 +124,13 @@ private MesosCluster(String clusterId) {
...
                         break;
                     case "registrator":
-                        this.containers.add(new Registrator(dockerClient, this, uuid, containerId));
+                        this.containers.add(new Registrator(DockerClientFactory.get(), this, uuid, containerId));
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java]: * Does `Registrator` take docker client as a parameter?

File: minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java (hunk 1 above)
Comment (remmelt) [inline minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java]: 1) I forgot to remove the parameter there
2) get() -> build() OK

**Comment by sadovnikov** [comment]
Approved

**Comment by frankscholten** [comment]
LGTM

**Pull request merged**